common/setups/rasr/util/decode.py

```diff
@@ -0,0 +1,132 @@
+__all__ = [
+    "RecognitionParameters",
+    "PriorPath",
+    "SearchJobArgs",
+    "Lattice2CtmArgs",
+    "StmArgs",
+    "ScliteScorerArgs",
+    "OptimizeJobArgs",
+]
+
+from dataclasses import dataclass
+from itertools import product
+from typing import Any, Dict, List, Optional, Tuple, TypedDict, Union
+
+from sisyphus import tk
+
+import i6_core.rasr as rasr
+
+from ..config.am_config import Tdp
+
+
+class RecognitionParameters:
+    def __init__(
+        self,
+        *,
+        am_scales: List[float],
+        lm_scales: List[float],
+        prior_scales: List[float],
+        pronunciation_scales: List[Optional[float]],
```

Comment:
```suggestion
        pronunciation_scales: Optional[List[float]],
```
the list is optional, not the floats in the list